[Docs] Voyager v1.6 (#20006)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-voyager/reference/data-migration/export-data.md

    identifier: voyager-export-data
    parent: data-migration
    weight: 20
aliases:
  - /preview/yugabyte-voyager/reference/fall-forward/fall-forward-synchronize/
type: docs
rightNav:
  hideH4: true
---

The following page describes the following export commands:

- [export data](#export-data) (and export data from source)
- [export data status](#export-data-status)
- [get data-migration-report](#get-data-migration-report)
- [export data from target](#export-data-from-target)

> ### export data

Reviewer: Why bump heading for commands down to 3? Worthy of 2...

Author: I did that for right nav purpose. On second thoughts, it looks like I can bump it up as the syntax, arguments and example showing up may be useful. Ill change it.

#### Syntax

Syntax for export data is as follows:

```text
Usage: yb-voyager export data [ <arguments> ... ]
```

Syntax for export data from source is as follows:

```text
Usage: yb-voyager export data from source [ <arguments> ... ]
```

#### Arguments

The valid *arguments* for export data are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |
| --disable-pb |Use this argument to disable progress bar during data export and statistics printing during streaming phase. <br>Default: false<br> Accepted parameters: true, false, yes, no, 0, 1 |
| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to store exported schema DDL files, export data files, migration state, and a log file.|

> | --export-type | Choose migration type between `snapshot-only` (offline) or `snapshot-and-changes`(live, and optionally fall-forward). <br>Default: `snapshot-only` |

Reviewer: ```suggestion
| --export-type | Specify the migration type as `snapshot-only` (offline) or `snapshot-and-changes` (live, and optionally fall-forward). <br>Default: `snapshot-only` |
```

| [--source-ssl-crl](../../yb-voyager-cli/#ssl-connectivity) <path> | Path to a file containing the SSL certificate revocation list (CRL).|
| [--source-ssl-mode](../../yb-voyager-cli/#ssl-connectivity) <SSLmode> | One of `disable`, `allow`, `prefer`(default), `require`, `verify-ca`, or `verify-full`. |
| [--source-ssl-root-cert](../../yb-voyager-cli/#ssl-connectivity) <path> | Path to a file containing SSL certificate authority (CA) certificate(s). |
| --start-clean | Starts a fresh data export after clearing all data from the `data` directory. <br> Default: false <br> Accepted parameters: true, false, yes, no, 0, 1 |
| --table-list | Comma-separated list of the tables for which data needs to be exported. By default, table names in the list are case-insensitive. Enclose names in double quotes ("") to make them case-sensitive.<br>Table names can also be glob patterns containing wildcard characters, such as an asterisk (*) (matches zero or more characters) or question mark (?) (matches one character). To use a glob pattern for table names, enclose the list in single quotes ('').<br> For example, `--table-list '"Products", order*'`. |

> | --exclude-table-list <tableNames> | Comma-separated list of the tables for which data needs to be excluded during export. Table names follow the same convention as `--table-list`. |

Reviewer: some of these are out of order

> | --exclude-table-list <tableNames> | Comma-separated list of the tables for which data needs to be excluded during export. Table names follow the same convention as `--table-list`. |

Reviewer: ```suggestion
| --exclude-table-list <tableNames> | Comma-separated list of the tables to exclude during export. Table names follow the same convention as `--table-list`. |
```

### Syntax

Syntax for export data is as follows:

```text
Usage: yb-voyager export data [ <arguments> ... ]
```

Syntax for export data from source is as follows:

```text
Usage: yb-voyager export data from source [ <arguments> ... ]
```

### Arguments

The valid *arguments* for export data are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |
| --disable-pb |Use this argument to disable progress bar during data export and statistics printing during streaming phase. <br>Default: false<br> Accepted parameters: true, false, yes, no, 0, 1 |
| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to store exported schema DDL files, export data files, migration state, and a log file.|

> | --export-type | Specify the migration type as `snapshot-only` (offline) or `snapshot-and-changes` (live, and optionally fall-forward). <br>Default: `snapshot-only` |

Reviewer: optionally fall forward? you don't need to specify 'snapshot-and-changes' if you want to do fall forward, or fall back for that matter? not clear

Author: export-type is snapshot-and-changes for live migration for all types. So ill change it. Thanks

The valid *arguments* for get data-migration-report are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |
| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to store exported schema DDL files, export data files, migration state, and a log file.|
| -h, --help | Command line help. |
| --source-db-password <password>| Password to connect to the source database server. If you don't provide a password via the CLI during any migration phase, yb-voyager will prompt you at runtime for a password. Alternatively, you can also specify the password by setting the environment variable `SOURCE_DB_PASSWORD`. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose it in single quotes. |
| --source-replica-db-password <password> | Password to connect to the source-replica database server. Alternatively, you can also specify the password by setting the environment variable `SOURCE_REPLICA_DB_PASSWORD`. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose it in single quotes. |

> | --target-db-password <password>| Password to connect to the target YugabyteDB server. Alternatively, you can also specify the password by setting the environment variable `TARGET_DB_PASSWORD`. If you don't provide a password via the CLI during any migration phase, yb-voyager will prompt you at runtime for a password. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose the password in single quotes. |

Reviewer: target YugabyteDB server - wait, the database or the server?

Author: I am not seeing the usage of server in the cli help .. Will change it to database.

#### Example

```sh
yb-voyager get data-migration-report --export-dir /dir/export-dir
```

### export data from target

**Note** that this command is applicable for Live migrations only.

Exports new changes from the target YugabyteDB database to import to the source database (for fall-back migration)source-replica database (for fall-forward migration) so that the source/source-replica database can be in sync with the YugabyteDB database after cutover.

#### Syntax

```text
Usage: yb-voyager export data from target [ <arguments> ... ]
```

#### Arguments

The valid *arguments* for export data from target are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |

> | --disable-pb | Use this argument to disable progress bar during data export and printing statistics during the streaming phase. <br>Default false.<br> Accepted parameters: true, false, yes, no, 0, 1 |

Reviewer: ```suggestion
| --disable-pb | Use this argument to disable the progress bar during data export and printing statistics during the streaming phase. <br>Default: false<br> Accepted parameters: true, false, yes, no, 0, 1 |
```

> | --exclude-table-list <tableNames> | Comma-separated list of tables to exclude while exporting data. <br> Table names can include glob wildcard characters `?` (matches one character) and `*` (matches zero or more characters). In case the table names are case sensitive, double-quote them. For example, `--exclude-table-list 'orders,"Products",items'`. |

Reviewer: ```suggestion
| --exclude-table-list <tableNames> | Comma-separated list of tables to exclude while exporting data. <br> Table names can include glob wildcard characters `?` (matches one character) and `*` (matches zero or more characters). Enclose names in double quotes ("") to make them case-sensitive. For example, `--exclude-table-list 'orders,"Products",items'`. |
```

| --exclude-table-list-file-path | Path of the file containing comma-separated list of table names to exclude while exporting data. |
| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to store exported schema DDL files, export data files, migration state, and a log file.|
| -h, --help | Command line help for synchronize. |
| --send-diagnostics | Enable or disable sending [diagnostics](../../../diagnostics-report/) information to Yugabyte. <br>Default: true<br> Accepted parameters: true, false, yes, no, 0, 1 |

> | --table-list | Comma-separated list of the table names to export data. Table names can include glob wildcard characters `?` (matches one character) and `*` (matches zero or more characters). In case the table names are case sensitive, double-quote them. For example `--table-list 'orders,"Products",items'`. |

Reviewer: match text to other table list text

| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to store exported schema DDL files, export data files, migration state, and a log file.|
| -h, --help | Command line help. |

#### Example

```sh
yb-voyager export data status --export-dir /dir/export-dir
```

### get data-migration-report

Provides a consolidated report of data migration per table among all the databases involved in the live migration. The report includes the number of rows exported, the number of rows imported, change events exported and imported (INSERTS, UPDATES, and DELETES), and the final row count on the database.

> 

Reviewer 2: Add in the starting, `For live migration only, it provides. a consolidated ...`

Author: Adding a general note at the beginning of each live migration command on export and import data cli pages .

| --source-replica-db-password <password> | Password to connect to the source-replica database. Alternatively, you can also specify the password by setting the environment variable `SOURCE_REPLICA_DB_PASSWORD`. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose it in single quotes. |
| --target-db-password <password>| Password to connect to the target YugabyteDB database. Alternatively, you can also specify the password by setting the environment variable `TARGET_DB_PASSWORD`. If you don't provide a password via the CLI during any migration phase, yb-voyager will prompt you at runtime for a password. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose the password in single quotes. |

### Example

```sh
yb-voyager get data-migration-report --export-dir /dir/export-dir
```

## export data from target

**Note** that this command is applicable for Live migrations only.

> Exports new changes from the target YugabyteDB database to import to the source database (for fall-back migration)source-replica database (for fall-forward migration) so that the source/source-replica database can be in sync with the YugabyteDB database after cutover.

Reviewer: slightly mangled
